apps/server/src/modules/user/legacy/service/admin-users.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain/types';
+import { Logger } from '@src/core/logger';
+import { User } from '@shared/domain/entity';
+import { EntityNotFoundError } from '@shared/common';
+import { UsersAdminRepo } from '../repo';
+import { UserListResponse, UserResponse, UsersSearchQueryParams } from '../controller/dto';
+
+@Injectable()
+export class AdminUsersService {
+	constructor(private readonly usersAdminRepo: UsersAdminRepo, private readonly logger: Logger) {
+		this.logger.setContext(AdminUsersService.name);
+	}
+
+	async getUsersWithNestedData(
+		roleId: string | undefined,
+		schoolId: EntityId,
+		schoolYearId: EntityId | undefined,
+		params: UsersSearchQueryParams
+	): Promise<UserListResponse> {
+		const usersResponse = (await this.usersAdminRepo.getUsersWithNestedData(
+			roleId,
+			schoolId,
+			schoolYearId,
+			params
+		)) as UserListResponse[];
```

Comment:
Normally I would say that we don't want to have a type assertion here but a mapper. But since the code is in this legacy-folder, I don't know... Has this been clarified with @CeEv?

Reply:
The reason here is entityManager.aggregate returns any[] where inside there is this particular object.